File: Java/HF Design Patterns/Factory/src/patterns/abstractfactory/pizzas/Pizza.java
package patterns.abstractfactory.pizzas;

import patterns.abstractfactory.ingredients.*;

public abstract class Pizza {
    Dough dough;
    Sauce sauce;
    Cheese cheese;
    Clams clams;

    public abstract void prepare();

    public void bake(){

    }
    public void cut(){

    }
    public void box(){
        System.out.println(sauce.getClass().getName());
    }
}

File: CSharp/AbstractFactory/AbstractFactory/Pizza.cs
using AbstractFactory.Ingredients;

namespace AbstractFactory
{
    public class Pizza
    {
        private ICheese cheese;

        public Pizza(IPizzaRecipe recipe) {
            this.cheese = recipe.GetCheese();
        }

         public string GetAllIngredients() {
            return cheese.GetName();
         }
    }
}